src/java/jni.rs

```diff
@@ -997,4 +999,132 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn check_gen_jni_function_namespace_and_cfg() {
+        let func_name = "testDummyFuncName";
+        let native_name = "test_dummy_func_name";
+        let mut ctx = Context::default();
+        let mut dummy_outputs: HashMap<String, String> = HashMap::new();
+        let dummy_func_str = indoc!(
+        "/// Comments are dumb, we don't need them here.
+        #[cfg(feature = \"mock-routing\")]
+        #[no_mangle]
+        pub unsafe extern \"C\" fn test_dummy_func_name(
+            dummy_app: *mut App,
+            dummy_user_data: *mut c_void,
+            dummy_o_cb: extern \"C\" fn(dummy_user_data: *mut c_void, dummy_result: *const FfiResult),
+        ) {}"
+        );
+        let test_str = "# [ cfg ( feature = \"mock-routing\" ) ]\n\
+        # [ no_mangle ] \
+        pub unsafe extern \"system\" fn Java_net_maidsafe_dummy_NativeBindings_testDummyFuncName ( \
+            env : JNIEnv , \
+            _class : JClass , \
+            dummy_app : jlong , \
+            dummy_user_data : JObject , \
+            dummy_o_cb : JObject \
+        ) { \
+        let dummy_app = dummy_app as * mut App ; \
+        let dummy_user_data = jni_unwrap ! ( c_void :: from_java ( & env , dummy_user_data ) ) ; \
+        let ctx = gen_ctx ! ( env , dummy_o_cb ) ; \
+        test_dummy_func_name ( dummy_app , & dummy_user_data , ctx , call_CallbackVoidFfiResult ) ; }"
+        ;
+        let file: syn::File = unwrap!(syn::parse_str(dummy_func_str));
+        for item in file.items {
+            if let syn::Item::Fn(ref func) = item {
+                let attri = &func.attrs[..];
+                let inputs: Vec<_> = func.decl.inputs.iter().cloned().collect();
+                let string = generate_jni_function(
```

Comment:
Maybe use a more descriptive name here? E.g.
```suggestion
                let generated_jni_code = generate_jni_function(
```